spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/DefaultRowWriter.scala

```diff
@@ -19,13 +19,12 @@ class DefaultRowWriter[T : TypeTag : ColumnMapper](
 
   override def readColumnValues(data: T, buffer: Array[Any]) = {
     val row = converter.convert(data)
-    for (i <- columnNames.indices)
+    for (i <- row.columnValues.indices)
```

Comment:
This returns values in the order of the data and not in the order of the selected columns?

Reply:
makes sense